crypto/rand/rand_lcl.h

```diff
@@ -17,21 +17,24 @@
 # include <openssl/ec.h>
 # include "internal/rand.h"
 
-/* we require 256 bits of randomness */
-# define RANDOMNESS_NEEDED (256 / 8)
+/* Amount of randomness (in bytes) we want for initial seeding. */
+# define RANDOMNESS_NEEDED              (128 / 8)
```

Comment:
No it doesn't, for several reasons:
- The value that was changed is not used for the NIST DRBG, it has it's own value for what the minimum amount of entropy is needed based on the DRBG that is instantiated. We can have an AES128, AES192 or AES256 DRBG. And each can support a number of security strengths from the list of 112, 128, 192 and 256. We're actually missing parameters in our instantiating function to say at which security level we want to work. If we'd want (the DRNG) to work at the 256 security level, we'd need the 256 bits of entropy.
- For the NIST DRBG, since none of our sources are "full entropy" sources, but have been processed, if you follow the NIST SP 800-90 documents we need to double the amount entropy. That is, if you want to reach the 128  bit security strength you need 256 bits of entropy from a source like /dev/urandom.
- If a cipher uses AES, we prefer AES256 over AES128. 

But this doesn't say anything at all about any relation between the security strength of the RNG and the key you want to generate with it. I don't actually know enough about this, but the RFC you pointed at seems to suggest that the amount of entropy needs to be double that of your key size.

It also seems to me that if you only have 2^128 possible inputs, you can only generate 2^128 possible 256 bit keys, which seems to defeat the point of generating a 256 bit key.